packages/grid/_modules_/grid/models/params/gridCellParams.ts

```diff
@@ -52,13 +55,16 @@ export interface GridCellParams {
    * @param id
    * @param field
    */
-  getValue: (id: GridRowId, field: string) => GridCellValue;
+  getValue: (id: GridRowId, field: string) => T;
 }
 
 /**
  * GridCellParams containing api.
  */
-export interface GridRenderCellParams extends GridCellParams {
+export interface GridRenderCellParams<
+  T = any,
```

Comment:
```suggestion
  T = GridCellValue,
```

Reply:
If I apply this, typescript complains for `renderCell` callbacks, see https://github.com/mui-org/material-ui-x/pull/2436#discussion_r694634035 thread for more details. 

But if we keep `any` type, then [tests fail](https://app.circleci.com/pipelines/github/mui-org/material-ui-x/7753/workflows/aeec0160-2c39-4738-a417-36eab53d5890/jobs/42873) for `GridActionsCell` `getActions` after regenerating propTypes for `formattedValue` and `value` fields. 

What should we do? Should we make `formattedValue` and `value` optional in `GridActionsCellProps` props?